ambry-replication/src/test/java/com.github.ambry.replication/BlobIdTransformerTest.java

```diff
@@ -381,6 +475,35 @@ private Message buildMessage(String blobIdString, Class clazz, boolean hasEncryp
     private InputStream createBlobStream() {
       return new ByteArrayInputStream(randomByteArray(BLOB_STREAM_SIZE));
     }
+
+    private ByteBuffer createMetadataByteBuffer(String... datachunkIds) throws IOException {
+      List<StoreKey> storeKeys = new ArrayList<>();
+      for (String datachunkId : datachunkIds) {
+        storeKeys.add(blobIdFactory.getStoreKey(datachunkId));
+      }
+      ByteBuffer output = MetadataContentSerDe.serializeMetadataContent(4000000, 8000000, storeKeys);
+      output.flip();
+      return output;
+    }
+
+    private ByteBuffer createInputMetadataByteBuffer() throws IOException {
+      return createMetadataByteBuffer(BLOB_ID_PAIR_VERSION_3_CONVERTED.getFirst(),
+          BLOB_ID_PAIR_VERSION_2_CONVERTED.getFirst());
+    }
+
+    private ByteBuffer createBrokenDeprecatedMetadataByteBuffer() throws IOException {
```

Comment:
doc on all these functions. For e.g. the name of the function doesn't describe what "broken" means

Reply:
refactored to take out methods